src/class-handler.php

```diff
@@ -0,0 +1,143 @@
+<?php
+/**
+ * Duplicate Post handler class for duplication actions.
+ *
+ * @package Duplicate_Post
+ * @since 4.0
+ */
+
+namespace Yoast\WP\Duplicate_Post;
+
+/**
+ * Represents the handler for duplication actions.
+ */
+class Handler {
+
+	/**
+	 * Post_Duplicator object
+	 *
+	 * @var Post_Duplicator
+	 */
+	private $post_duplicator;
+
+	/**
+	 * Initializes the main class.
+	 *
+	 * @param Post_Duplicator $post_duplicator The Post_Duplicator object.
+	 */
+	public function __construct( Post_Duplicator $post_duplicator ) {
+		$this->post_duplicator = $post_duplicator;
+		$this->register_hooks();
+	}
+
+	/**
+	 * Adds hooks to integrate with WordPress.
+	 *
+	 * @return void
+	 */
+	private function register_hooks() {
+		\add_action( 'admin_action_duplicate_post_copy_for_rewrite', [ $this, 'rewrite_link_action_handler' ] );
+		\add_action( 'admin_init', [ $this, 'add_bulk_handlers' ] );
+	}
+
+	/**
+	 * Handles the action for copying a post for the Rewrite and Republish feature.
+	 *
+	 * @return void
+	 */
+	public function rewrite_link_action_handler() {
+		if ( ! \duplicate_post_is_current_user_allowed_to_copy() ) {
+			\wp_die( \esc_html__( 'Current user is not allowed to copy posts.', 'duplicate-post' ) );
+		}
+
+		if ( ! ( isset( $_GET['post'] ) || isset( $_POST['post'] ) || // Input var okay.
+			( isset( $_REQUEST['action'] ) && 'duplicate_post_copy_for_rewrite' === $_REQUEST['action'] ) ) ) { // Input var okay.
+			\wp_die( \esc_html__( 'No post to duplicate has been supplied!', 'duplicate-post' ) );
+		}
+
+		$id = ( isset( $_GET['post'] ) ? intval( \wp_unslash( $_GET['post'] ) ) : intval( \wp_unslash( $_POST['post'] ) ) ); // Input var okay.
```

Comment:
I think `intval` should be backslashed? Not 100% sure, I'm asking :)

Reply:
Since it's a standard PHP function, I'd say no (like we don't do it to `isset()` or `empty()`)